rest/route/task_restart.go

```diff
@@ -10,13 +11,16 @@ import (
 	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/evergreen/rest/model"
 	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/utility"
 	"github.com/pkg/errors"
 )
 
 // taskRestartHandler implements the route POST /task/{task_id}/restart. It
 // fetches the needed task and project and calls the service function to
 // set the proper fields when reseting the task.
 type taskRestartHandler struct {
+	FailedOnly *bool `json:"FailedOnly"`
```

Comment:
I changed it back to a pointer because after experimenting with it being not a pointer, the JSON was not reading it, I will experiment again with it not being a pointer since I have a firmer grasp of it!

Reply:
I just implemented it without the pointer and it works as intended with parsing still! It removed a lot of bloat as well